crypto/ec/ecp_nistp384.c

```diff
@@ -0,0 +1,1931 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * A 64-bit implementation of the NIST P-384 elliptic curve point multiplication
+ *
+ * Designed for 56-bit limbs by Rohan McLure <rmclure@linux.ibm.com>.
+ * The layout is based on that of ecp_nistp{224,521}.c, allowing even for asm
+ * acceleration of felem_{square,mul} as supported in these files.
+ */
+
+#include <openssl/e_os2.h>
+
+#include <string.h>
+#include <openssl/err.h>
+#include "ec_local.h"
+
+#include "internal/numbers.h"
+
+#ifndef INT128_MAX
+# error "Your compiler doesn't appear to support 128-bit integer types"
+#endif
+
+typedef uint8_t u8;
+typedef uint64_t u64;
+
+/*
+ * The underlying field. P384 operates over GF(2^384-2^128-2^96+2^32-1). We
+ * can serialize an element of this field into 48 bytes where the most
+ * significant byte contains only a single bit. We call this an felem_bytearray.
```

Comment:
Is the 'Most significant byte contains only a single bit' correct? (That would be 377 bits?)

Reply:
Apologies, yes this comment is completely wrong. I got this confused with `felem_reduce`, which returns a 385 bit number.